File: PracAux121/Polimorfismo/Ejercicio2/Videojuego.java
import java.util.Scanner;

public class Videojuego {
    public String nombre;
    public String plataforma;
    public int cantidadDeJugadores;
    public Videojuego(){
        this.nombre="God of War";
        this.plataforma="Playstation";
        this.cantidadDeJugadores=0;
    }
    public Videojuego(String nombre, String plataforma){
        this.nombre=nombre;
        this.plataforma=plataforma;
        this.cantidadDeJugadores=0;
    }
    public Videojuego(String nombre, String plataforma, int cantidadDeJugadores){
        this.nombre=nombre;
        this.plataforma=plataforma;
        this.cantidadDeJugadores=cantidadDeJugadores;
    }
    public void agregarJugadores(int x){
        x=1;
        this.cantidadDeJugadores+=x;
    }
    public void agregarJugadores(){
        Scanner lee=new Scanner(System.in);
        System.out.println(" Ingrese el numero de jugadores que agregara: ");
        int cantidad = lee.nextInt();
        this.cantidadDeJugadores+=cantidad;
    }

    @Override
    public String toString() {
        return "[ el nombre del juego es: "+this.nombre+", su platafomra es: "+this.plataforma+", y la cantidad de jugadores es: "+this.cantidadDeJugadores+" ]";
    }
    public static void main(String[]args){
        System.out.println("------------");
        System.out.println("Videojuego 1");
        System.out.println("------------");
        Videojuego videojuego1=new Videojuego();
        videojuego1.agregarJugadores(1);
        System.out.println(videojuego1);
        System.out.println("------------");
        System.out.println("Videojuego 2");
        System.out.println("------------");
        Videojuego videojuego2=new Videojuego("Mario Kart","Nintendo");
        videojuego2.agregarJugadores();
        System.out.println(videojuego2);
        System.out.println("------------");
        System.out.println("Videojuego 3");
        System.out.println("------------");
        Videojuego videojuego3=new Videojuego("League of legends","PC",400000);
        videojuego3.agregarJugadores();
        System.out.println(videojuego3);
    }
}

File: PracAux121/Polimorfismo/Ejercicio2/Videojuego.py
class Videojuego:
    def __init__(self, nombre=None, plataforma=None, cantidad_jugadores=0):
        if nombre is None and plataforma is None:
               self.nombre="Desconocido"
               self.plataforma="Desconocido"
               self.cantidad_jugadores=0
        elif plataforma is None:
               self.nombre=nombre
               self.plataforma="Desconocida"
               self.cantidad_jugadores=0
        else:
               self.nombre=nombre
               self.plataforma=plataforma
               self.cantidad_jugadores=cantidad_jugadores
    def agregarJugadores(self, cantidad=None):
        if cantidad is None:
                self.cantidad_jugadores+=1
        else:
            self.cantidad_jugadores+=cantidad
    def __str__(self):
        return f"Videojuego: {self.nombre}, Plataforma: {self.plataforma}, Jugadores: {self.cantidad_jugadores}"
    
juego1= Videojuego()
juego2= Videojuego("FIFA 23")
juego3=Videojuego("Mario Kart", "Nintendo Switch", 24)

juego1.agregarJugadores()
juego2.agregarJugadores(17)

print(juego1)
print(juego2)
print(juego3)